src/socket/address.cr

```diff
@@ -213,13 +252,23 @@ class Socket
     end
 
     def to_s(io)
+      if abstract?
+        io << '@'
+      end
       io << path
     end
 
     def to_unsafe : LibC::Sockaddr*
       sockaddr = Pointer(LibC::SockaddrUn).malloc
       sockaddr.value.sun_family = family
-      sockaddr.value.sun_path.to_unsafe.copy_from(@path.to_unsafe, @path.bytesize + 1)
+
+      destination = sockaddr.value.sun_path.to_unsafe
+      if @abstract
+        destination[0] = 0_u8
+        destination += 1
+      end
+      destination.copy_from(@path.to_unsafe, @path.bytesize + 1)
```

Comment:
This line is the reason you should prefer Slice: Your write to `destination` is bound to the size of `@path`, and not checked to the size of `destination`. You get this check for free with Slice.

Reply:
Thanks @Papierkorb !
What about `@path` should I keep the `#to_unsafe`? (I think it's not harmful here, but maybe there's a better way I didn't see?)

Also, I can't remember/can't find what this `+ 1` in `@path.bytesize + 1` for the count is for, it seems to work without it, soo... If there's no objection I'll remove it with the `.to_slice` fix soon.

Reply:
> Also, I can't remember/can't find what this + 1 in @path.bytesize + 1 for the count is for, it seems to work without it

Looks like it was intended to copy the trailing `NUL`. You don't "need" it right now cause the call to `Pointer#malloc` above asks BoehmGC to allocate it, and BoehmGC zeros memory for you. Thus, you should either 1) Copy the trailing `NUL` 2) or set it manually. Not doing so makes that code rely on an implementation detail of the current GC implementation.

Reply:
The `@path.to_unsafe` could be replaced with `String#to_slice`. I'm not sure if the returned slice contains the trailing `NUL`, please verify that.

Reply:
@Papierkorb the returned slice doesn't contain the null - why would it? Slice is a crystal type and crystal doesn't use null terminated strings.